### PR TITLE
added prow jobs for dependency-watchdog

### PR DESCRIPTION
**Pull request by unmarshall**
/kind enhancement

**What this PR does / why we need it**:
This PR adds prow jobs for unit tests and image builds of dependency-watchdog repository.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by rfranzke** [comment]
@oliver-goetz Do we want/are we ready to let further repositories be managed via Prow? IIRC, last time we decided to only manage g/g and a few extensions until we have more maintainers. Is this still up-to-date?

**Review by oliver-goetz — CHANGES_REQUESTED** [review]
There are some basic labels and plugins required for prow to work for a repository.
Please check what Dimitar did in `plugins.yaml`, `labels.yaml` and `ci-infra-periodics.yaml` for enabling `gardener-extension-shoot-oidc-service` see PR https://github.com/gardener/ci-infra/pull/157

**Comment by oliver-goetz** [comment]
> @oliver-goetz Do we want/are we ready to let further repositories be managed via Prow? IIRC, last time we decided to only manage g/g and a few extensions until we have more maintainers. Is this still up-to-date?

There is only very maintenance effort for us when there are only a couple of prow jobs sharing the same pattern we use for gardener/gardener. That's the same setup as `gardener-extension-shoot-oidc-service` Dimitar added a while ago.

There will be more effort when other repositories would like to enable the full prow functionality including tide and other github integrations. For this we might need more maintainers. Maybe Dimitar, Madhav or other people from Bangalore team could help, when we come to this point.

**Review by oliver-goetz — CHANGES_REQUESTED** [review]
There are too many labels and plugins for the "testing only" scenario in prow.

File: config/prow/labels.yaml
Comment (unmarshall) [reply config/prow/labels.yaml:2410]: Ok what was not clear for me is what is maintained by gardener-robot and what will be maintained by prow. For now i will just use the labels that `gardener-extension-shoot-oidc-service` defined then.

File: config/prow/plugins.yaml
Comment (unmarshall) [reply config/prow/plugins.yaml:239]: Ok this is not clear. Wouldn't we still require `approve` or `assign` or `lgtm` etc.?

Comment (unmarshall) [reply config/prow/plugins.yaml:239]: done

File: config/prow/labels.yaml
Comment (unmarshall) [reply config/prow/labels.yaml:2410]: done

**Review by oliver-goetz — APPROVED** [review]
/lgtm

**Pull request merged**
